Rivet/Functions/Checkpoint-Migration.ps1

```diff
@@ -0,0 +1,64 @@
+
+function Checkpoint-Migration
+{
+    <#
+    .SYNOPSIS
+    Checkpoints the current state of the database so that it can be re-created.
+
+    .DESCRIPTION
+    The `Checkpoint-Migration` function captures the state of a database after all migrations have been applied. The
+    captured state is exported to a `schema.ps1` file that can be applied with Rivet to re-create that state of the
+    database. Before capturing the current state, a Rivet Push is invoked to ensure all migrations have been applied.
+
+    .EXAMPLE
+    Checkpoint-Migration -Database $Database -Environment $Environment -ConfigFilePath $ConfigFilePath
+
+    Demonstrates how to checkpoint a migration.
+    #>
+    [CmdletBinding()]
+    param(
+        # The database(s) to migrate.
+        [Parameter(Mandatory)]
+        [String[]] $Database,
+
+        # The environment you're working in.
+        [String] $Environment,
+
+        # The path to the Rivet configuration file. Default behavior is to look in the current directory for a `rivet.json` file.
+        [String] $ConfigFilePath,
+
+        # The output path for the schema.ps1 file. If not provided the path will default to the build root.
+        [String] $OutputPath,
+
+        # If a schema.ps1 script already exists at the output path it will be overwritten when Force is given.
+        [Switch] $Force
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    if( ($OutputPath -and -not $Force) -and ( Test-Path -Path $OutputPath -PathType Leaf) )
+    {
+        Write-Error "A schema.ps1 file already exists at: $OutputPath. This file can be overwritten if given the `$Force switch."

```

Comment:
You're assuming the file name is always `schema.ps1` when it may be different if an explicit path was passed in to `$OutputPath`. Change this error message to:

```powershell
Write-Error "Checkpoint output path ""$($OutputPaht)"" already exists. Use the -Force switch to overwrite."
```